### PR TITLE
googleapis_auth: support GIS browser auth with new APIs

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by ditman** [comment]
Commenting because I cannot add myself as a reviewer.

**Comment by kevmoo** [comment]
...any comments @ditman ?

File: googleapis_auth/lib/src/oauth2_flows/token_model.dart [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+/// for more details.
+Future<AccessCredentials> requestAccessCredentials({
+  required String clientId,
+  required Iterable<String> scopes,
+  String prompt = 'select_account',
+  @Deprecated('Undocumented feature. Do not include in production code.')
+      String? logLevel,
+}) async {
+  await initializeScript('https://accounts.google.com/gsi/client');
+  if (logLevel != null) _googleAccountsId.callMethod('setLogLevel', [logLevel]);
```
Comment (ditman) [inline googleapis_auth/lib/src/oauth2_flows/token_model.dart:35]: Can you check `if (debug && logLevel != null)`?

File: googleapis_auth/lib/src/oauth2_flows/token_model.dart [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      return;
+    }
+
+    final token = AccessToken(
+      response.token_type,
+      response.access_token,
+      expiryDate(response.expires_in),
+    );
+
+    final creds = AccessCredentials(token, null, response.scope.split(' '));
```
Comment (ditman) [inline googleapis_auth/lib/src/oauth2_flows/token_model.dart:58]: Should it be documented that these access credentials will never contain an refreshToken, nor idToken/user profile information anymore?

File: googleapis_auth/lib/src/oauth2_flows/token_model_interop.dart [hunk 3]
```diff
@@ -0,0 +1,115 @@
+// ignore_for_file: non_constant_identifier_names
+
+@JS('google.accounts.oauth2')
```
Comment (ditman) [inline googleapis_auth/lib/src/oauth2_flows/token_model_interop.dart:3]: This looks great, I might steal it from you :P

**Review by ditman — APPROVED** [review]
LGTM, the logging call in google.accounts.id was a little bit confusing (why not make that part of the interop file, it should be ~2 lines of extra code there :P)

File: googleapis_auth/lib/src/oauth2_flows/token_model.dart (hunk 1 above)
Comment (kevmoo) [reply googleapis_auth/lib/src/oauth2_flows/token_model.dart:35]: Which `debug`? Where does `debug` come from?

File: googleapis_auth/lib/src/oauth2_flows/token_model.dart (hunk 2 above)
Comment (kevmoo) [reply googleapis_auth/lib/src/oauth2_flows/token_model.dart:58]: ack

File: googleapis_auth/lib/src/oauth2_flows/token_model.dart (hunk 1 above)
Comment (ditman) [reply googleapis_auth/lib/src/oauth2_flows/token_model.dart:39]: Nevermind, I was thinking of Flutter's `kDebugMode` :)

**Comment by kevmoo** [comment]
> LGTM, the logging call in google.accounts.id was a little bit confusing (why not make that part of the interop file, it should be ~2 lines of extra code there :P)

Because that library is mapped to a different "namespace" - `@JS('google.accounts.oauth2')`

**Pull request merged**